### PR TITLE
Media Uploading - HMAC error on JP connected sites

### PR DESCRIPTION
**Pull request by daniloercoli**
This is a *temporary?* fix for #402

This patch does check the `error` field available in the REST Response, and set the response error type to `REQUEST_TOO_LARGE ` if  `error` is equals to one of the two predefined error messages we get in case of  too large media.

A better fix is probably required server side.

### CODE REVIEW

**Comment by daniloercoli** [comment]
cc @SergioEstevao

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/media/MediaRestClient.java [hunk 1]
```diff
@@ -373,6 +373,18 @@ private MediaError parseUploadError(Response response) {
...
                 mediaError.message = body.getString("message");
             }
+
+            if (!siteModel.isWPCom()) {
+                // TODO : temporary fix for "big" media uploads on Jetpack connected site
+                // See https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/402
+                // Tried to upload a media that's too large (larger than the site's max_upload_filesize)
+                if (body.has("error")) {
+                    String error = body.getString("error");
+                    if ("invalid_hmac".equals(error) || "unknown".equals(error)) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/media/MediaRestClient.java]: I can see that we get an `unknown : Unknown error communicating with the Jetpack site` error if we exceed both `upload_max_filesize` and `post_max_size`, and I guess that's why you're changing the error for `unknown` too. But I'm not sure that we should make this change for `unknown` errors, as we could get those legitimately for other reasons (we could get the HMAC one too I guess, but that's less likely).

Comment (daniloercoli) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/media/MediaRestClient.java]: Yes, I was torn about catching "unknown" errors. I did some tests on JP sites, and there are times where the error reported is `HMAC....`, and other where it's `Unknown`. All depends on the hosting configuration.

Anyway, I removed the "unknown" catching and left the hmac only, that should be safe to catch. 
Ready for another pass @aforcier

**Comment by aforcier** [comment]
Heads up @daniloercoli this is failing due to a support library conflict - merging in current `develop` should fix that up.

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**